src/index.ts

```diff
@@ -83,6 +83,18 @@ extendConfig((config: HardhatConfig, userConfig: Readonly<HardhatUserConfig>) =>
     }
 });
 
+// copy all user-defined axios settings
+extendConfig((config: HardhatConfig, userConfig: Readonly<HardhatUserConfig>) => {
+    if (userConfig.axios) {
+        config.axios = JSON.parse(JSON.stringify(userConfig.axios));
+    }
+    if (!config.axios) {
+        config.axios = {
```

Comment:
Two things:
- Giving this an implementation specific name is not a good idea, it should be more abstract/generic
- To preserve namespace independence between various hardhat plugins, I'd recommend making this piece of settings a part of the `starknet` config object.

That being said, perhaps we could use an option called `httpTimeout` or `requestTimeout`.

Reply:
changed to starknet.requestTimeout